lib/core/unittest/CDataFrameTest.cc

```diff
@@ -429,7 +429,7 @@ BOOST_FIXTURE_TEST_CASE(testMemoryUsage, CTestFixture) {
     //   1) 1075 + 4 times the root directory length bytes for on disk, and
     //   2) data size + doc ids size + 900 byte overhead in main memory.
     std::size_t maximumMemory[]{1075 + 4 * rootDirectory.length(),
-                                rows * (cols + 1) * 4 + 900};
+                                rows * (4 * ((cols + 3) / 4) + 1) * 4 + 900};
```

Comment:
It might be worth adding a comment here to say the rounding up is due to alignment padding.  In a couple of years time it won't be clear to everyone why this formula is this complex.

Reply:
Added in [d0155db](https://github.com/elastic/ml-cpp/pull/1156/commits/d0155db8ceb24d9d46c4e6214df44551051c5add).